lib/AST/Attr.cpp

```diff
@@ -374,6 +374,13 @@ static std::string getDifferentiationParametersClauseString(
       case ParsedAutoDiffParameter::Kind::Self:
         printer << "self";
         break;
+      case ParsedAutoDiffParameter::Kind::Ordered:
+        auto *paramList = function->getParameters();
+        assert(param.getOrder() <= paramList->size() &&
```

Comment:
This shouldn't happen given that I check this [here](https://github.com/apple/swift/pull/25594/files#diff-163f1b115cbb43d2e7bcc01e6c5eb2ceR2895), but good to be safe 😄